spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/config/GatewayAutoConfiguration.java

```diff
@@ -582,6 +583,9 @@ else if (ssl.isUseInsecureTrustManager()) {
 								.trustManager(InsecureTrustManagerFactory.INSTANCE);
 					}
 
+					sslContextBuilder.keyManager(new File(ssl.getKeyCertChainPath()),
```

Comment:
There should probably be some checks for empty values

Reply:
Working on this

Reply:
Pushed changes for null check